You are currently using **version 1.5** notebook .  
The next cell imports Pandas package
then assigns the read DataFrame into variable df

In [5]:
import pandas as pd

df=pd.read_csv("WA_Sales_Products_2012-14.csv") # opens the csv file and assigns it to a DataFrame df
                #data is from https://community.watsonanalytics.com/wp-content/uploads/2015/08/WA_Sales_Products_2012-14.csv
df.head() #returns the first five rows



Retailer country Order method type  Retailer type       Product line  \
0    United States               Fax  Outdoors Shop  Camping Equipment   
1    United States               Fax  Outdoors Shop  Camping Equipment   
2    United States               Fax  Outdoors Shop  Camping Equipment   
3    United States               Fax  Outdoors Shop  Camping Equipment   
4    United States               Fax  Outdoors Shop  Camping Equipment   

    Product type                    Product  Year  Quarter    Revenue  \
0   Cooking Gear  TrailChef Deluxe Cook Set  2012  Q1 2012   59628.66   
1   Cooking Gear     TrailChef Double Flame  2012  Q1 2012   35950.32   
2          Tents                  Star Dome  2012  Q1 2012   89940.48   
3          Tents               Star Gazer 2  2012  Q1 2012  165883.41   
4  Sleeping Bags            Hibernator Lite  2012  Q1 2012  119822.20   

   Quantity  Gross margin  
0       489      0.347548  
1       252      0.474274  
2       147      0.352772  
3       303      0.282938  
4      1415      0.291450

What are the column labels of this data ?

In [6]:
df.columns



Index(['Retailer country', 'Order method type', 'Retailer type',
       'Product line', 'Product type', 'Product', 'Year', 'Quarter', 'Revenue',
       'Quantity', 'Gross margin'],
      dtype='object')

How many rows and columns are used ?

In [3]:
df.shape #returns the rows and columns count



(88475, 11)

What is the current index what type of variable is it?

In [8]:
df.index #the index is an object that contains 0...to...88474



RangeIndex(start=0, stop=88475, step=1)

In [9]:
type(df.index) #it is a class pandas.core.indexes.range.RangeIndex



pandas.core.indexes.range.RangeIndex

Of the 88475 rows, what is the total quantity of products on this file ?

In [54]:
df['Quantity'].sum()



69062361

Of the 69,062,361 total quantity, what is the breakdown per country ?

In [57]:
df.groupby(['Retailer country'])['Quantity'].sum()



Retailer country
Australia          2000781
Austria            2102206
Belgium            1799021
Brazil             1957452
Canada             4419144
China              4596986
Denmark             962099
Finland            2786800
France             4044378
Germany            3733677
Italy              2852400
Japan              5137596
Korea              2932353
Mexico             2547117
Netherlands        2687404
Singapore          2849009
Spain              2527950
Sweden             1403596
Switzerland        1822191
United Kingdom     4050932
United States     11849269
Name: Quantity, dtype: int64

Of the 69,062,361 total quantity, what is the breakdown per Quarter ?

In [58]:
df.groupby(['Year', 'Quarter'])['Quantity'].sum()



Year  Quarter
2012  Q1 2012    6258427
      Q2 2012    5579643
      Q3 2012    5556853
      Q4 2012    6129762
2013  Q1 2013    6094787
      Q2 2013    6783135
      Q3 2013    6467989
      Q4 2013    6595879
2014  Q1 2014    8382882
      Q2 2014    8344594
      Q3 2014    2868410
Name: Quantity, dtype: int64

What are the total quatities by Country, by Year and by Quarter ?

In [51]:
df.set_index(['Retailer country','Year','Quarter' ] ).groupby(['Retailer country','Year','Quarter' ])['Quantity'].sum()



Retailer country  Year  Quarter
Australia         2012  Q1 2012     136096
                        Q2 2012     121604
                        Q3 2012     120750
                        Q4 2012     143736
                  2013  Q1 2013     189677
                        Q2 2013     223759
                        Q3 2013     217266
                        Q4 2013     209048
                  2014  Q1 2014     275275
                        Q2 2014     267777
                        Q3 2014      95793
Austria           2012  Q1 2012     186218
                        Q2 2012     166415
                        Q3 2012     172302
                        Q4 2012     177199
                  2013  Q1 2013     212131
                        Q2 2013     194371
                        Q3 2013     185057
                        Q4 2013     212212
                  2014  Q1 2014     243716
                        Q2 2014     262506
                        Q3 2014      90079
Belgium           2012

The above examples show some of Pandas features used in extracting summaries formats and figures given so many details as in a medium-sized table.  

